dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04200CreateDefaultVanityURL.java

```diff
@@ -0,0 +1,241 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This upgrade task will insert into the tables Inode and Structure the default Vanity URL Content
+ * Type. This is needed to be able to migrate the old Vanity URL's (on the virtual_link table) to
+ * the new Vanity URL that are content.
+ *
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Jun 6, 2017
+ */
+public class Task04200CreateDefaultVanityURL extends AbstractJDBCStartupTask {
+
+    /* If the Content Type is changed we need to check the values here are the same that the ones on the Starter */
+    private static final String DEFAULT_VANITY_URL_STRUCTURE_INODE = "8e850645-bb92-4fda-a765-e67063a59be0";
+    private static final String DEFAULT_VANITY_URL_STRUCTURE_VARNAME = "Vanityurlasset";
+    private static final String DEFAULT_VANITY_URL_STRUCTURE_NAME = "Vanity URL Asset";
+    private static final String DEFAULT_VANITY_URL_STRUCTURE_DESCRIPTION = "Vanity URL as Content";
+    private static final String[] DEFAULT_VANITY_URL_FIELDS_INODES = {
+            "4e64a309-8c5e-48cf-b4a9-e724a5e09575", // Order
+            "3b7f25bb-c5be-48bb-b00b-5b1b754e550c", // Action
+            "de4fea7f-4d8f-48eb-8a63-20772dced99a", // Foward To
+            "7e438b93-b631-4812-9c9c-331b03e6b1cd", // Uri
+            "49f3803a-b2b0-4e03-bb2e-d1bb2a1c135e", // Site
+            "9ae3e58a-75c9-4e4a-90f3-6f0d5f32a6f0"  // Title
+    };
+
+    private static final String INSERT_INODE_QUERY = "insert into inode (inode, owner, idate, type) values (?, ?, ?, ?)";
+
+    private static final String INSERT_FIELD_QUERY = "insert into field "
+            + "(inode,structure_inode,field_name,field_type,field_relation_type,field_contentlet,"
+            + "required,indexed,listed,velocity_var_name,sort_order,field_values,regex_check,"
+            + "hint,default_value,fixed,read_only,searchable,unique_,mod_date) "
+            + "VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+    private static final String INSERT_STRUCTURE_QUERY =
+            "insert into structure (inode, name, description, structuretype, system, fixed, velocity_var_name, host, "
+                    + "folder, mod_date, default_structure) "
+                    + "values ( ?, ?, ?, ?, ?, ?, ?, ?, ? ,?,?)";
+
+    private static final int INT_7 = 7;
+    private static final int INT_6 = 6;
+    private static final int INT_5 = 5;
+    private static final int INT_4 = 4;
+    private static final int INT_3 = 3;
+    private static final int INT_2 = 2;
+    private static final int INT_1 = 1;
+    private static final String TEXT_FIELD = "com.dotcms.contenttype.model.field.TextField";
+    private static final String CUSTOM_FIELD ="com.dotcms.contenttype.model.field.CustomField";
+    private static final String SELECT_FIELD = "com.dotcms.contenttype.model.field.SelectField";
+
+    @Override
+    public void executeUpgrade() throws DotDataException, DotRuntimeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)